### PR TITLE
Comments Processor

### PR DESCRIPTION
**Pull request by jamuraa**
I've hacked up the Comments Processor.  I didn't hook it up to the ListView, so I'm not sure it works, but this is one way to handle the parameters.  I would expect that it is more common for parameters to exist than not, so it makes sense to put them into the ResourceProcessor interface.

I tried to use the existing framework that was there where possible, but I might have missed something.  Let me know!

### CODE REVIEW

File: src/mn/aug/restfulandroid/service/CatPictureCommentsProcessorFactory.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+public class CatPictureCommentsProcessorFactory {
+
+	private static CatPictureCommentsProcessorFactory mSingleton;
+	private ResourceProcessor mProcessor;
+
+	public void setDefaultProcessor(ResourceProcessor processor) {
+		mProcessor = processor;
+	}
+
+	public ResourceProcessor getProcessor() {
```
Comment (aug-mn) [inline src/mn/aug/restfulandroid/service/CatPictureCommentsProcessorFactory.java]: This should take a resource type and return the correct processor for that type.

**Comment by aug-mn** [comment]
Awesome Michael thanks for submitting our first pull request!  There's a change I want to make to the processor factory.  Pre-existing, but seeing your code reminded me that it needed to be refactored.  There should only be one processor factory,  It should take in a resource type and return the correct processor to handle that specific resource.  Having a factory per resource (that only returns one thing) doesn't make too much sense, othe rthan to maybe encapsulate a complex build strategy, which we don't have.
Anyhow, is that something you'd care to take on and resubmit?

**Comment by jamuraa** [comment]
Yes, I'll make those changes (and remove some of the random pasted text that my flaky touchpad added too) and resubmit this.  I was somewhat confused at the factory as well - I thought it might be used to "embed" the parameters into the resourceprocessor, but after rethinking it makes more sense to pass them into the methods of the ResourceProcessor.

**Comment by jamuraa** [comment]
There's more refactoring to do here, I think we could probably do some fancy footwork to remove the switch() from the DefaultCatPicturesService.  How does it look now?

**Comment by aug-mn** [comment]
Hi Michael, sorry for the delay, just realized I wasn't getting emails from this account, all fixed now!
So these changes look good.  Seeing your refactoring I realize we can totally remove the switch from the service, it serves no purpose right now.  I'll accept these commits and make the change.  Thanks!!

**Pull request merged**